store/postgres/searchmapper.go

```
@@ -0,0 +1,105 @@
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"mono/store"
+
+	// Register some standard stuff
+	_ "github.com/lib/pq"
+)
+
+// SearchMapper ...
+type SearchMapper struct {
+	db *sql.DB
+}
+
+// SearchUser ...
+func (sm *SearchMapper) SearchUser(user store.User) (bool, error) {
+
+	ctx := context.Background()
+	tx, err := sm.db.BeginTx(ctx, nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// insert user to user_events
+	userQuery := `INSERT INTO user_status_events (user_id, version, status)
+	VALUES ($1, $2, 'searching')
+	`
+	_, err = tx.ExecContext(ctx, userQuery, user.ID, user.Version)
+	if err != nil {
+		// Incase we find any error in the query execution, rollback the transaction
+		tx.Rollback()
+		return false, fmt.Errorf("something went wrong")
+	}
+
+	// search another user in users events
+	// insert user to user_events
+	matchQuery := `SELECT user_id, version FROM user_status_events 
+	WHERE status = 'searching' and user_id IS NOT $1
+	ORDER BY updated_at
+	DESC
+	`
+
+	row := tx.QueryRow(matchQuery, user.ID)
+	var foundUser store.User
+	// Store the count in the `catCount` variable
+	err = row.Scan(&foundUser)
+	if err != nil {
+		tx.Rollback()
+		return false, err
+	}
+
+	// Found
+	if foundUser.ID != 0 {
+		userQuery := `UPDATE user_status_events  SET version = $2 , status = 'busy'
+		WHERE user_id = $1`
+
+		newVersion := user.Version + 1
+		_, err = tx.ExecContext(ctx, userQuery, user.ID, newVersion)
+		if err != nil {
+			// Incase we find any error in the query execution, rollback the transaction
+			tx.Rollback()
+			return false, fmt.Errorf("something went wrong")
+		}
+
+		newVersion = foundUser.Version + 1
+		_, err = tx.ExecContext(ctx, userQuery, foundUser.ID, newVersion)
+		if err != nil {
+			// Incase we find any error in the query execution, rollback the transaction
+			tx.Rollback()
+			return false, fmt.Errorf("something went wrong")
+		}
+	}
+
+	err = tx.Commit()
+	if err != nil {
+		log.Fatal(err)
```

Fatal terminates the process. Maybe we just want to log the error here?

I suggest to handle the errors on the top level layer (not even logging them here).
We can return the error with a key word like "postgres". 
This would lead us to the following:
`return false, fmt.Sprintf("postgres SearchUser: %v", err)`